app/src/androidTest/kotlin/at/bitfire/davdroid/db/migration/Migration18Test.kt

```diff
@@ -70,6 +71,53 @@ class AutoMigration18Test : DatabaseMigrationTest(toVersion = 18) {
         }
     )
 
+    @Test
+    fun testMigration_RemovesOrphanSyncstats() = testMigration(
+        prepare = { db ->
+            // Insert a valid service and one valid collection
+            db.execSQL(
+                "INSERT INTO service (id, accountName, type) VALUES (?, ?, ?)",
+                arrayOf<Any?>(1, "test", 1)
+            )
+            db.execSQL(
+                "INSERT INTO collection (id, serviceId, url, type, privWriteContent, privUnbind, forceReadOnly, sync) " +
+                        "VALUES (?, ?, ?, ?, ?, ?, ?, ?)",
+                arrayOf<Any?>(1, 1, "https://example.com/1", 1, 1, 1, 0, 1)
+            )
+
+            // Valid syncstats row (referencing existing collection)
+            db.execSQL(
+                "INSERT INTO syncstats (id, collectionId, authority, lastSync) VALUES (?, ?, ?, ?)",
+                arrayOf<Any?>(1, 1, "com.android.contacts", 1234)
+            )
+
+            // Two orphan syncstats rows (collectionId not in collection)
+            db.execSQL(
+                "INSERT INTO syncstats (id, collectionId, authority, lastSync) VALUES (?, ?, ?, ?)",
+                arrayOf<Any?>(2, 99, "com.android.calendar", 1660521600)
+            )
```

Comment:
How can that work when there is a `ForeignKey(childColumns = arrayOf("collectionId"), entity = Collection::class, parentColumns = arrayOf("id"))`?

Is there something wrong with the foreign key definition?

Reply:
Right. I will try to verify that and add an elaborate comment

Reply:
It should always have been enabled in production - so I think we can enable it to make sure and if it does cause foreign-key exceptions then we know that it was not properly enforced before. At least we can be (quite) certain then :)

Reply:
> It should always have been enabled in production,

I think so too. But how could it be disable in the original issue?

Maybe the user has edited the database manually previously with some tool that didn't set the PRAGMA?

@steelman Is it possible that you edited the database manually already some time before?